public/scripts/home.js

```diff
@@ -23,45 +30,82 @@ class RootViewModel {
 }
 
 function handleSearchInput(viewModel) {
```

Comment:
I see the search related style are put at search.css. May be all search related code should move to search.js?

Also, I see we have `scripts/algolia`. Why don't we put those into that folder, but scatter everywhere?

Reply:
> May be all search related code should move to search.js?

> Why don't we put those into that folder, but scatter everywhere?

I made the decision because the search bar behavior is not entirely the same for all below

1. search bar at home page (in `home.js`)
2. search bar at device list page header, when screen width >= 992px (in `device.js`)
3. search bar at device list page top, when screen width < 992px (in `device.js`)

I did consider putting all above 3 at one place `search.js` (just as I reused the styles in `search.css`), but I realized there are below difference between these 3

1. homepage only have 1 search bar
2. device list page have 2 search bars, which requires `querySelectAll`.

This, IMO, boils down to the question of `abstraction vs duplication`
I chose duplication here.
if we abstract `home.js` & `device.js` code into `search.js`, we would need to pass below parameters

```
containerIds: number[]

// then no matter what, loop through every containerId, with querySelectorAll
```

If further spec changes of homepage vs devicelist search bar, the abstraction would need more and more parameters.

Hence, I chose duplication here